traffic_ops/testing/api/v4/cachegroups_parameters_test.go

```diff
@@ -79,6 +90,129 @@ func CreateTestCacheGroupParameters(t *testing.T) {
 	testData.CacheGroupParameterRequests = append(testData.CacheGroupParameterRequests, resp.Response...)
 }
 
+func CreateTestCacheGroupParametersAlreadyExist(t *testing.T) {
+
+	resp, _, _ := TOSession.GetAllCacheGroupParameters(client.RequestOptions{})
+	cachegroupParameters := resp.Response
+	getAllcgparameters := cachegroupParameters.CacheGroupParameters
+	if len(getAllcgparameters) < 1 {
+		t.Fatalf("No existing cachegroup parameters available to test duplicate functionality")
+	}
+	if getAllcgparameters[0].Parameter == nil && *getAllcgparameters[0].Parameter != 0 {
+		t.Fatalf("found nil/0 Value in Paramters")
+	}
+	if len(*getAllcgparameters[0].CacheGroup) < 1 {
```

Comment:
This line will segfault if Traffic Ops incorrectly responded with a null or undefined Cache Group - also, I'm not sure why it's checking the length of that string. Yes an empty name is invalid, but it won't cause a segfault and will be caught later because the client call on line 113 will return an empty array

Reply:
Fixed it.